Source/WebKit/UIProcess/skia/BackingStoreSkia.cpp

```diff
@@ -29,34 +29,93 @@
 #if USE(SKIA)
 
 #include "UpdateInfo.h"
-#include <WebCore/NotImplemented.h>
+#include <WebCore/GraphicsContextSkia.h>
+#include <skia/core/SkBitmap.h>
+#include <skia/core/SkCanvas.h>
 
 namespace WebKit {
 
-static const Seconds s_scrollHysteresisDuration { 300_ms };
+static sk_sp<SkSurface> createSkiaImageSurfaceWithFastMalloc(const WebCore::IntSize& size, double deviceScaleFactor)
+{
+    ASSERT(!size.isEmpty());
+    WebCore::IntSize scaledSize(size);
+    scaledSize.scale(deviceScaleFactor);
+    auto info = SkImageInfo::MakeN32Premul(scaledSize.width(), scaledSize.height(), SkColorSpace::MakeSRGB());
+    auto* pixels = fastZeroedMalloc(scaledSize.height() * info.minRowBytes());
+    return SkSurfaces::WrapPixels(info, pixels, info.minRowBytes(), nullptr);
```

Comment:
The data is leaked. Since skia will already use bmalloc, I think it would be simpler to use `SkSurfaces::Raster` instead of `SkSurfaces::WrappPixels`, and skia will own the pixels.